apps/prairielearn/src/lib/server-jobs.js

```diff
@@ -46,9 +47,15 @@ const sql = sqldb.loadSqlEquiv(__filename);
   instead.
 */
 
-// Store currently active job information in memory. This is used
-// to accumulate stderr/stdout, which are only written to the DB
-// once the job is finished.
+/** @typedef {{ output: string }} AbstractJob */
+
+/**
+ * Store currently active job information in memory. This is used
+ * to accumulate stderr/stdout, which are only written to the DB
+ * once the job is finished.
+ *
+ * @type {Record<string, AbstractJob>}
```

Comment:
This type ensures that anything reading from this object doesn't rely on this being either a new or legacy server job. As long as it has an `output` property, this code is happy!

Reply:
Sneaky!